### PR TITLE
Fix HttpWebRequest on UWP and Xbox One (case 989588)

### PR DESCRIPTION
**Pull request by TautvydasZilys**
Don't fail constructing WebProxyData if retrieving proxy data from the registry fails with DllNotFoundException. We don't construct it on Unity AOT profile anyway, and it's better for it to work without proxy support than not work at all.

Release notes:

• Universal Windows Platform: Fixed HttpWebRequest throwing DllNotFoundException on IL2CPP scripting backend (case 989588).

### CODE REVIEW

**Comment by TautvydasZilys** [comment]
With these changes WebRequestTests in IL2CPP repo pass.

File: mcs/class/System/ReferenceSources/AutoWebProxyScriptEngine.cs [hunk 1]
```diff
@@ -153,6 +164,56 @@ WebProxyData InitializeRegistryGlobalProxy ()
 			return null;
 		}
 
+		WebProxyData InitializeProxyDataUWP()
```
Comment (TautvydasZilys) [inline mcs/class/System/ReferenceSources/AutoWebProxyScriptEngine.cs]: This proves I need to read my diffs before commiting. /facepalm. I'll remove this, it was a WIP copy-pasta.

**Comment by joshpeterson** [comment]
Can you add release notes to the PR description and the commit message? Should this be back ported?

**Comment by TautvydasZilys** [comment]
It would be nice to backport it to 2018.1, but it doesn't have to necessarily make into RC (it's not a shipstopper). I'm not sure what process Mono path to trunk and backporting follows - could you summarize it? I'll put release notes in the description.

**Comment by joshpeterson** [comment]
To back port a change, just cherry-pick the commit to the `unity-2018.1-mbe` branch.

**Comment by TautvydasZilys** [comment]
Ok, thanks!
